src/libraries/System.Text.Json/gen/Reflection/ReflectionExtensions.cs

```diff
@@ -45,5 +46,26 @@ private static bool HasJsonConstructorAttribute(ConstructorInfo constructorInfo)
 
             return false;
         }
+
+        public static Location? GetDiagnosticLocation(this Type type)
+        {
+            TypeWrapper? typeWrapper = type as TypeWrapper;
```

Comment:
nit\suggestion: using a cast instead of `as` would give you a more debuggable exception (here and below). You could probably skip the Assert then too.

Reply:
+1 although I might still express this as follows:
```
Debug.Assert(type is TypeWrapper);
return (TypeWrapper)type.Location;
```
For a better debugging experience.

Reply:
Applied the feedback, thanks.